internal/backend/state.go

```diff
@@ -168,7 +168,11 @@ func (state *State) Rename(ctx context.Context, oldName, newName string) error {
 			if exists, err := tx.Mailbox.Query().Where(mailbox.Name(superior)).Exist(ctx); err != nil {
 				return err
 			} else if exists {
-				continue
+				if superior == oldName {
+					return ErrExistingMailbox
+				} else {
+					continue
```

Comment:
Idiomatic go style advice: if you `return` inside the `if` block, don't use an `else` block (since the code that follows will only be executed when the condition was not met).